RubeesEat/Model/DB/DbPersonRepository.cs

```diff
@@ -27,24 +40,53 @@ public IReadOnlyList<Person> GetAll()
 
     public void Add(Person person)
     {
-        using var connection = connectionFactory.CreateDbConnection();
-        using var command = connection.CreateCommand("INSERT INTO Persons (PersonID, FirstName, LastName) VALUES (@PersonID, @FirstName, @LastName);");
+        using var connection = _connectionFactory.CreateDbConnection();
+        using var command = connection.CreateCommand("INSERT INTO Persons (PersonID, FirstName, LastName, LoginName) VALUES (@PersonID, @FirstName, @LastName, @LoginName);");
         command.AddParameter("@PersonID", person.Id);
         command.AddParameter("@FirstName", person.FirstName);
         command.AddParameter("@LastName", person.LastName);
+        command.AddParameter("@LoginName", person.LoginName);
         
         connection.Open();
         command.ExecuteNonQuery();
     }
     
     public Person GetCurrentUser()
     {
-        return GetAll()[^1];
+        var httpContext = _httpContextAccessor.HttpContext;
+        if (httpContext == null)
+            throw new NullReferenceException("HttpContext is null");
+        var loginName = httpContext.User.FindFirst("name")?.Value;
+        var nickName = httpContext.User.FindFirst("nickname")?.Value;
+        var splitNickName = nickName.Split(".");
+        var firstName = splitNickName[0];
+        var lastName = splitNickName[1];
+        if (loginName == null)
+            throw new NullReferenceException("User is inactive");
+
+        using var connection = _connectionFactory.CreateDbConnection();
+        using var command = connection.CreateCommand("SELECT PersonID, FirstName, LastName FROM Persons WHERE LoginName = @LoginName;");
+        command.AddParameter("@LoginName", loginName);
+
+        connection.Open();
+        using var reader = command.ExecuteReader();
+        if (reader.Read())
+        {
+            return new Person
+            (
+                reader.GetGuid(0),
+                reader.GetString(1),
+                reader.GetString(2)
+            );
+        }
+        var newPerson = new Person(Guid.NewGuid(), firstName, lastName, loginName);
+        Add(newPerson);
+        return newPerson;
```

Comment:
Please extract this into its own method `Person? GetByLoginName(string loginName)` we can then share the mapping logic with `GetById`